files/default/archive-resource-revisions.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+DB_URL=$(grep 'sqlalchemy\.url' /etc/ckan/default/production.ini |awk -F '://' '{print $2}')
+export PGUSER=$(echo "$DB_URL" |awk -F: '{print $1}')
+DB_URL=$(echo "$DB_URL" |awk -F: '{print $2}')
+export PGPASSWORD=$(echo "$DB_URL" |awk -F@ '{print $1}')
+DB_URL=$(echo "$DB_URL" |awk -F@ '{print $2}')
+export PGHOST=$(echo "$DB_URL" |awk -F/ '{print $1}')
+export PGDATABASE=$(echo "$DB_URL" |awk -F/ '{print $2}')
+
+psql -f $(dirname $0)/create-resource-revision-archival-table.sql || echo "Archival table already exists or could not be created"
+psql -1f $(dirname $0)/archive-resource-revisions.sql
```

Comment:
will this not delete if the insert fails?

do you need to have `commit;`  at the end of the script?

Reply:
That is the purpose of the `-1` flag.

       -1
       --single-transaction
           When psql executes a script, adding this option wraps BEGIN/COMMIT around the script to execute it as a single transaction.
           This ensures that either all the commands complete successfully, or no changes are applied.

           If the script itself uses BEGIN, COMMIT, or ROLLBACK, this option will not have the desired effects. Also, if the script
           contains any command that cannot be executed inside a transaction block, specifying this option will cause that command (and
           hence the whole transaction) to fail.